apollo-router/src/plugins/fleet_detector.rs

```diff
@@ -232,6 +241,128 @@ impl PluginPrivate for FleetDetector {
             *store = GaugeStore::active(&self.gauge_options);
         }
     }
+
+    fn router_service(&self, service: router::BoxService) -> router::BoxService {
+        if !self.enabled {
+            return service;
+        }
+
+        service
+            // Count the number of request bytes from clients to the router
+            .map_request(move |req: router::Request| router::Request {
+                router_request: req.router_request.map(move |body| {
+                    router::Body::wrap_stream(body.map(move |res| {
```

Comment:
We probably just need https://docs.rs/futures/latest/futures/prelude/stream/trait.StreamExt.html#method.inspect